src/main/java/org/apache/commons/io/FileUtils.java

```diff
@@ -1603,93 +1598,272 @@ public static boolean isFileNewer(final File file, final File reference) {
     }
 
     /**
-     * Tests if the specified <code>File</code> is newer than the specified
-     * time reference.
+     * Tests if the specified {@code File} is newer than the specified time reference.
      *
-     * @param file       the <code>File</code> of which the modification date must
-     *                   be compared, must not be {@code null}
+     * @param file       the {@code File} of which the modification date must be compared
      * @param timeMillis the time reference measured in milliseconds since the
      *                   epoch (00:00:00 GMT, January 1, 1970)
-     * @return true if the <code>File</code> exists and has been modified after
-     * the given time reference.
-     * @throws IllegalArgumentException if the file is {@code null}
+     * @return true if the {@code File} exists and has been modified after the given time reference.
+     * @throws NullPointerException if the file is {@code null}
      */
     public static boolean isFileNewer(final File file, final long timeMillis) {
-        if (file == null) {
-            throw new IllegalArgumentException("No specified file");
-        }
+        Objects.requireNonNull(file, "file");
         if (!file.exists()) {
             return false;
         }
         return file.lastModified() > timeMillis;
     }
 
     /**
-     * Tests if the specified <code>File</code> is older than the specified
-     * <code>Date</code>.
+     * Tests if the specified {@code File} is newer than the specified {@code Instant}.
+     *
+     * @param file    the {@code File} of which the modification date must be compared
+     * @param instant the date reference
+     * @return true if the {@code File} exists and has been modified after the given {@code Instant}.
+     * @throws NullPointerException if the file or instant is {@code null}
+     *
+     * @since 2.8
+     */
+    public static boolean isFileNewer(final File file, final Instant instant) {
```

Comment:
Curious here...what do you think happens if the TimeZone of the local datetime of the machine on which the code is running differs from the `TimeZone` embedded in the Instant. I worry that `instant.toEpochMilli();` might return the `Epoch` from the timezone of the instant which could be off by multiples of 30 minutes, and thus could yield differing results depending upon a user error here. Maybe we should convert the instant to the TimeZone of the machine before running `instant.toEpochMilli();`??

Reply:
@Isira-Seneviratne Would you please reply to @chtompki ?

Reply:
@garydgregory My bad, I'll do so now.

@chtompki How should the conversion be done?

Reply:
This is not true, an Instant never has a timezone. Unfortunately this method now goes on the forbiddenAPI list of Lucene because of this.
Instant is by definition not related to any time zone. toEpochMillies returns a UNIX timestamp.

Reply:
The big risk of this forward/backward transformation is now that at the DST changes hour, the calculation is wrong: During DST change the same local date time exists 2 times. When you convert the instant to a local time (after local time went backwards due to DST change) and then you convert the local time back, you're off by an hour.
The correct way here is to just call toEpochMillies(). PLEASE DO NOT CONVERT FORWARD BACKWARDS from to/local time.

Reply:
Because of this the whole methods added here are not useable in international application, except the one taking an explicit ZoneId. The one which would have been correct (the one taking an Instant) was corrupted by this forward/backward transformation.